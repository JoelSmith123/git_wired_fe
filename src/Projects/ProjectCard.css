

.ProjectCard {
  margin:  3%;
  padding: 3%;
  width:   30%;
  height:  auto;
  background-color: #BDCCD8;

  display:         flex;
  flex-direction:  column;
  flex-wrap:       wrap;
  justify-content: center;
  align-items:     stretch;

  text-align: center;
}

.ProjectTitle {
  font-size:   150%;
  color:       black;
  font-weight: bolder;
  text-align:  center;
}

.RepoName {
  font-size:  85%;
  color:      #454545;
  text-align: center;
  /* font-weight: bolder; */
}

.Updated {
  font-size:   85%;
  color:       #454545;
  font-weight: bolder;
  text-align:  center;
}

.Created {
  font-size:   70%;
  color:       #454545;
  text-align:  center;
  /* font-weight: bolder; */
}
