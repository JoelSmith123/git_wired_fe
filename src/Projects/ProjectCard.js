import React, { Component } from 'react';
import './ProjectCard.css'


export default class ProjectCard extends Component {

  // --- Project Name ---

  getTitle() {
    return this.props.project['projectTitle']
  }

  renderTitle(){
    return (
      <div className='ProjectTitle'>{ this.getTitle() }</div>
    )
  }

  // --- Repo Name ---

  getRepo() {
    return this.props.project['repoName']
  }

  renderRepo(){
    return (
      <div className='RepoName'> { this.getRepo() } </div>
    )
  }

  // --- Updated ---

  getUpdated() {
    return this.props.project['updatedAt']
  }

  renderUpdated(){
    return (
      <div className='Updated'> { this.getUpdated() } </div>
    )
  }

  // --- Created ---

  getCreated() {
    return this.props.project['createdAt']
  }

  renderCreated(){
    return (
      <div className='Created'> { this.getCreated() } </div>
    )
  }


  // --- Image ---

  // TO DO - Display image (not MVP)



  render() {
    return (
      <span className='ProjectCard'
            onClick= { () => { this.props.changeViewPageState('wireframe')  }  }>
        { this.renderTitle()   }
        { this.renderRepo()    }
        { this.renderUpdated() }
        { this.renderCreated() }
      </span>
    )
  }


}
