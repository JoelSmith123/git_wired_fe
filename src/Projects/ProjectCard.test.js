import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import ProjectCard from './ProjectCard.js'



describe('ProjectCard', () => {
  let mockProject
  beforeEach(() => {
    mockProject = {
      'projectTitle': 'Title1',
      'repoName':     'RepoName1',
      'createdAt':    'Created1',
      'updatedAt':    'Updated1',
    }
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ProjectCard project={mockProject} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<ProjectCard project={mockProject} />)
    expect(wrapper).toMatchSnapshot()
  }) 

  describe('Title', () => {
    it('gets project title', () => {
      const wrapper = shallow(<ProjectCard project={ mockProject }/>)
    
      expect(wrapper.instance().getTitle()).toEqual(mockProject.projectTitle)      
    });

    it('renders project title', ()=> {
      const wrapper = shallow(<ProjectCard project={ mockProject }/>)

      expect(wrapper.instance().renderTitle()).toEqual(<div className="ProjectTitle">{wrapper.instance().getTitle()}</div>)
    });
  });

  describe('Repo Name', () => {
    it('gets repo name', () => {
      const wrapper = shallow(<ProjectCard project={ mockProject }/>)
    
      expect(wrapper.instance().getRepo()).toEqual(mockProject.repoName)
    });

    it('renders repo name', () => {
      const wrapper = shallow(<ProjectCard project={ mockProject }/>)

      expect(wrapper.instance().renderRepo()).toEqual(<div className="RepoName">{wrapper.instance().getRepo()}</div>)
    });
  });

  describe('Created At', () => {
    it('gets updated at', () => {
      const wrapper = shallow(<ProjectCard project={ mockProject }/>)
    
      expect(wrapper.instance().getUpdated()).toEqual(mockProject.updatedAt)
    });

    it('renders updated at', () => {
      const wrapper = shallow(<ProjectCard project={ mockProject }/>)

      expect(wrapper.instance().renderUpdated()).toEqual(<div className="Updated">{wrapper.instance().getUpdated()}</div>)
    });
  });

  describe('Updated At', () => {
    it.skip('gets updated at', done => {

      // TO DO - TEST HERE

      done();
    });
    it.skip('renders updated at', done => {

            // TO DO - TEST HERE

      done();
    });
  });

  it('renders card', done => {

    // TO DO - TEST HERE

    done();
  });


});
