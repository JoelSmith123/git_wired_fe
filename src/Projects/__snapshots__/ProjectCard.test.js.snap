// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectCard
    project={
      Object {
        "createdAt": "Created1",
        "projectTitle": "Title1",
        "repoName": "RepoName1",
        "updatedAt": "Updated1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ProjectTitle"
        >
          Title1
        </div>,
        <div
          className="RepoName"
        >
          RepoName1
        </div>,
        <div
          className="Updated"
        >
           
          Updated1
           
        </div>,
        <div
          className="Created"
        >
           
          Created1
           
        </div>,
      ],
      "className": "ProjectCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title1",
          "className": "ProjectTitle",
        },
        "ref": null,
        "rendered": "Title1",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "RepoName1",
          "className": "RepoName",
        },
        "ref": null,
        "rendered": "RepoName1",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "Updated1",
            " ",
          ],
          "className": "Updated",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Updated1",
          " ",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "Created1",
            " ",
          ],
          "className": "Created",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Created1",
          " ",
        ],
        "type": "div",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ProjectTitle"
          >
            Title1
          </div>,
          <div
            className="RepoName"
          >
            RepoName1
          </div>,
          <div
            className="Updated"
          >
             
            Updated1
             
          </div>,
          <div
            className="Created"
          >
             
            Created1
             
          </div>,
        ],
        "className": "ProjectCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title1",
            "className": "ProjectTitle",
          },
          "ref": null,
          "rendered": "Title1",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "RepoName1",
            "className": "RepoName",
          },
          "ref": null,
          "rendered": "RepoName1",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "Updated1",
              " ",
            ],
            "className": "Updated",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Updated1",
            " ",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "Created1",
              " ",
            ],
            "className": "Created",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Created1",
            " ",
          ],
          "type": "div",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
